version: '3'

volumes:
  postgres_volume:
  redis-data:

services:
  auth_svc:
    build:
      context: .
      dockerfile: Dockerfile
    restart: always
    env_file:
      - local.env
    environment:
      POSTGRES_PORT: 5432
      POSTGRES_HOST: auth_db
    ports:
      - "50050:50051"
    networks:
      - auth_nw
      - cluster

  auth_db:
    image: postgres:14-alpine
    volumes:
      - postgres_volume:/var/lib/postgresql/data
    env_file:
      - local.env
    ports:
      - "6430:5432"
    networks:
      - auth_nw
      - cluster

  auth_cache:
    image: redis:7.2.5-alpine
    ports:
      - "6370:6379"
    volumes:
      - redis-data:/data
    networks:
      - auth_nw
      - cluster

  migrator:
    build:
      context: .
      dockerfile: migrator.Dockerfile
    restart: on-failure
    env_file:
      - local.env
    environment:
      DB_HOST: auth_db
    networks:
      - auth_nw

networks:
  auth_nw:
  cluster:
    external: true
